test/eureka-client.test.js

```diff
@@ -554,6 +555,42 @@ describe('Eureka client', () => {
     });
   });
 
+  describe('addInstanceMetadata()', () => {
+
+    let client, config, metadataSpy;
+    beforeEach(() => {
+      config = {
+        instance: {
+          app: 'app', vipAddress: '1.2.3.4', port: 9999, dataCenterInfo: { name: 'Amazon'},
+          statusPageUrl: 'http://__HOST__:8080/',
+          healthCheckUrl: 'http://__HOST__:8077/healthcheck',
+        },
+        eureka: {host: '127.0.0.1', port: 9999}
+      };
+      client = new Eureka(config);
+      metadataSpy = sinon.spy();
+
+      //let awsClientStub = sinon.createStubInstance(AwsMetadata);
+      sinon.stub(AwsMetadata.prototype, 'fetchMetadata').yields({
```

Comment:
You should probably stub the method on the instance rather than in the prototype.